accounts/models.py

```diff
@@ -3,19 +3,56 @@
 from django.contrib.auth import get_user_model
 from django.contrib.auth.models import AbstractUser
 from django.db import models
+from phonenumber_field.modelfields import PhoneNumberField
 
 
 class User(AbstractUser):
     # implicit username, email, first_name, and last_name fields
     # from AbstractUser that contains the user's PennKey
     pennid = models.IntegerField(primary_key=True)
     uuid = models.UUIDField(unique=True, default=uuid.uuid4, editable=False)
+    preferred_name = models.CharField(max_length=225, blank=True)
+
+    VERIFICATION_EXPIRATION_MINUTES = 10
+
+    def get_preferred_name(self):
+        if self.preferred_name != "":
+            return self.preferred_name
+        else:
+            return self.first_name
 
 
 class Student(models.Model):
     user = models.OneToOneField(get_user_model(), on_delete=models.DO_NOTHING)
     major = models.CharField(max_length=255, blank=True)
     school = models.CharField(max_length=255, blank=True)
+    graduation_year = models.IntegerField(null=True)
```

Comment:
Should this be blank instead of null? I never really got a good idea of that

Reply:
null=True allows null values in the previously existing rows in the db that don't have a value for graduation_year to be null I think (?), so if we don't do null=True we'll want to set default=something. blank=True makes this field optional?